pkg/kubelet/pod_workers.go

```diff
@@ -977,12 +999,16 @@ func (p *podWorkers) SyncKnownPods(desiredPods []*v1.Pod) map[types.UID]PodWorkT
 
 	p.podsSynced = true
 	for uid, status := range p.podSyncStatuses {
-		if _, exists := known[uid]; !exists {
+		if _, exists := known[uid]; !exists || status.restartRequested {
```

Comment:
Is it safe to removeTerminatedWorker here while status.restartRequested being set? After removeTerminatedWorker, the new podworker would be run, but the old podworker can be still in terminating, two goroutines may share the data struct like p.podUpdates[uid] and p.lastUndeliveredWorkUpdate[pod.UID].  I'm afraid there may be hidden deadlocks

Reply:
removeTerminatedWorker only removes if the worker is actually terminated.

Reply:
so the restartRequested knob on status will ensure that the worker is terminated, and the status mapping is then cleared for that pod uid, so restartRequested is a one-way flag and we dont need to reset state on it ever.  ignore my earlier comment/question.